frontend/src/tests/user-permission-tests.js

```diff
@@ -14,42 +14,142 @@ import { databaseSeeder } from "./setup";
+        // Set the active user to the target user
+        let resp = await apiPOST("/debug/active_user", newActiveUser);
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
+            `/debug/active_user`,
+            defaultUser
+        );
+        expect(respSwitchBackUser).toHaveStatus("success");
+    }
+
     beforeAll(async () => {
         await databaseSeeder.seed(api);
+        let resp = await apiGET(`/debug/active_user`);
+        expect(resp).toHaveStatus("success");
+        defaultUser = {
+            utorid: resp.payload.utorid,
+            roles: resp.payload.roles,
+        };
+
+        resp = await apiPOST("/debug/users", instructorOnlyUserData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiPOST("/debug/users", TAOnlyUserData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiPOST("/debug/users", TAInstructorUserData);
+        expect(resp).toHaveStatus("success");
     }, 30000);
 
     it("Admin user can access admin route", async () => {
         // the default user has all roles, including the admin role
-        const resp = await apiGET("/admin/active_user");
+        let resp = await apiGET("/admin/active_user");
         expect(resp).toHaveStatus("success");
     });
 
     it("A non-admin cannot access the admin route", async () => {
-        let resp = await apiPOST("/admin/users", newUserData);
-        // Save any extra attributes we got, such as an `id`
-        Object.assign(newUserData, resp.payload);
+        await switchToTargetUser(instructorOnlyUserData);
 
-        // Set the active user to the newly-created user
-        resp = await apiPOST("/debug/active_user", { id: newUserData.id });
+        // Try to fetch admin routes
+        let resp = await apiGET("/admin/active_user");
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiGET("/admin/instructors");
+        expect(resp).toHaveStatus("error");
 
-        // Try to fetch an admin route
-        resp = await apiGET("/admin/active_user");
+        resp = await apiGET("/admin/sessions");
         expect(resp).toHaveStatus("error");
 
-        // Set the active_user back to the default.
-        resp = await apiPOST("/debug/active_user", {
-            id: databaseSeeder.seededData.active_user.id,
-        });
+        resp = await apiGET("/admin/users");
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiGET("/admin/applicants");
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
+
+    it("An instructor can access instructor routes", async () => {
+        // the default user has all roles, including the instructor role
```

Comment:
Are we sure we're currently the default user? I would feel safer with a `restoreDefaultUser()` before this.